### ISSUE TITLE
window.Resize() does not work if SetFixedSize(true) is set after

### ISSUE THREAD

**Issue by FredPont** [fyne-io/fyne#2819]
### Describe the bug:
In Fyne 2.1.3 the order of  window.Resize() and SetFixedSize(true) matters. This is not the case in Fyne 2.1.2

### To Reproduce: 

This code only works on Fyne 2.1.2. On Fyne 2.1.3, it does not resize the window w which appears minimized :
```go
w.Resize(fyne.NewSize(finalWidth, finalHeight)) 
w.SetFixedSize(true)
```

Whereas this code works perfectly 
```go
w.SetFixedSize(true)
w.Resize(fyne.NewSize(finalWidth, finalHeight)) 
```

### Device :
 - **OS:**  Linux Manjaro 
 - **Version:**  up do date 2022-03-04
 - **Go version:**  go1.17.6 linux/amd64
 - **Fyne version:** 2.1.3

**Comment by kheyse-oqton** [comment]
May or may not be related to https://github.com/fyne-io/fyne/issues/2800, another window resize/fixedsize related regression since 2.0

**Comment by andydotxyz** [comment]
Thanks for connecting the two, but #2800 isn't a regression as far as I know, the initial size of an image using `FillOriginal` has always been off, it corrects on the second draw. Normal windows then adjust accordingly but FixedSize windows don't. It too needs fixed, but this regression is much higher priority.

**Comment by kheyse-oqton** [comment]
it is a regression because in 2.0 the same code works without problem. 
but we should probably discuss that on that issue.

**Comment by andydotxyz** [comment]
> * **OS:**  Linux Manjaro

I cannot replicate on macOS, this may be Linux specific which could explain why it was missed I guess.
Window sizing is one of the most complex code paths :(.

**Comment by andydotxyz** [comment]
I cannot replicate on Debian/FyneDesk either. Can you please provide more info about the failing WM and desktop setup @FredPont ?

**Comment by andydotxyz** [comment]
This may also depend on the content of the window I guess - the excerpt provided may not be enough to replicate.

**Comment by FredPont** [comment]
> I cannot replicate on Debian/FyneDesk either. Can you please provide more info about the failing WM and desktop setup @FredPont ?

@andydotxyz 

I did the test on 3 computers with XFE 4.16.
computer 1 & 2 Linux manjaro
computer 3 : Xubuntu 21.10

I reproduce the issue on the 3 computers.

Here are more information about one of my manjaro computer :

cat /etc/lsb-release
DISTRIB_ID=ManjaroLinux
DISTRIB_RELEASE=21.2.4
DISTRIB_CODENAME=Qonos
DISTRIB_DESCRIPTION="Manjaro Linux"


lspci -v
VGA compatible controller: NVIDIA Corporation GF119 [NVS 310] (rev a1) (prog-if 00 [VGA controller])
        Subsystem: NVIDIA Corporation Device 094e
        Physical Slot: 2
        Flags: bus master, fast devsel, latency 0, IRQ 92, NUMA node 0
        Memory at f2000000 (32-bit, non-prefetchable) [size=16M]
        Memory at e8000000 (64-bit, prefetchable) [size=128M]
        Memory at f0000000 (64-bit, prefetchable) [size=32M]
        I/O ports at 7000 [size=128]
        Expansion ROM at f3000000 [virtual] [disabled] [size=512K]
        Capabilities: <access denied>
        Kernel driver in use: nvidia
        Kernel modules: nouveau, nvidia_drm, nvidia


systemctl status display-manager
● lightdm.service - Light Display Manager
     Loaded: loaded (/usr/lib/systemd/system/lightdm.service; enabled; vendor preset: disabled)
     Active: active (running) since Fri 2022-03-04 16:03:29 CET; 2 days ago
       Docs: man:lightdm(1)
   Main PID: 1164 (lightdm)
      Tasks: 6 (limit: 77166)
     Memory: 51.2M
        CPU: 2.218s
     CGroup: /system.slice/lightdm.service
             ├─1164 /usr/bin/lightdm
             ├─1177 /usr/lib/Xorg :0 -seat seat0 -auth /run/lightdm/root/:0 -nolisten tcp vt7 -novtswitch
             └─1362 lightdm --session-child 12 19

**Comment by andydotxyz** [comment]
> I did the test on 3 computers with XFE 4.16.

Maybe it is specific to that WM?

Is it possible to replicate the problem with a minimal example, that will be very helpful.
For example a small modification to the hello world example?

**Comment by FredPont** [comment]
@andydotxyz ,

I send you a zip file with a minimal example to reproduce the issue and a video showing that, depending on the position of ` w.SetFixedSize(true)`, the fyne window is minimized or not.

https://user-images.githubusercontent.com/53525948/157188389-25f4102f-f978-4519-bf5d-a8b686a6645e.mp4

[debug.zip](https://github.com/fyne-io/fyne/files/8203709/debug.zip)

**Comment by andydotxyz** [comment]
Thanks, verified with this code - it seems that some content triggers it and others don't

**Comment by andydotxyz** [comment]
If you get a chance please test h#2829 - I think it resolves this issue

**Comment by andydotxyz** [comment]
Fixed in v2.1.4 under vote now
